src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPathContextProvider.cs

```diff
@@ -140,50 +125,54 @@ public bool TryCreateSolutionContext(out IVsPathContext outputPathContext)
         }
 
 
-        private static async Task<Dictionary<string, EnvDTE.Project>> GetPathToDTEProjectLookupAsync(EnvDTE.DTE dte)
+        private async Task<Tuple<IVsProjectAdapter, string>> CreateProjectAdapterAsync(string projectUniqueName)
```

Comment:
To avoid using a tuple as a return value and rely on comments to know what each tuple means, would it be better to change this method to `private async Task<bool> TryCreateProjectAdapterAsync(string projectUniqueName, out IVsProjectAdapter instance, out string projectAssetsFile)`?

Reply:
Unfortunately we can't have out parameters with async api which is why I had to fallback to tuple.